lib/schedule/data.ex

```diff
@@ -383,6 +383,8 @@ defmodule Schedule.Data do
         gtfs_data.all_modes.stop_times_by_trip_id
       )
 
+    version = :crypto.hash(:sha256, :erlang.term_to_binary(all_files))
```

Comment:
**suggestion (non-blocking):** I know this might be more of a pain to do implementation-wise, but if you hash the original HASTUS zip file that's downloaded it would probably be faster, as you wouldn't need to re-serialize the data with `term_to_binary`.

Reply:
(I'm also not actually sure if `term_to_binary` is guaranteed to return the same exact encoding when given the same term, but that's probably getting way off into nitpick territory.)

Reply:
> (I'm also not actually sure if term_to_binary is guaranteed to return the same exact encoding when given the same term, but that's probably getting way off into nitpick territory.)

Ooh you're right, I hadn't thought about that! From the [erlang docs](https://www.erlang.org/doc/man/erlang#term_to_binary-1):

> There is no guarantee that this function will return the same encoded representation for the same term.

I _think_ this is OK for this purpose, where `term_to_binary` will be called once on application startup. Though it does mean that users may be hit with a longer request than necessary when new instances of Skate spin up, or even if their request is routed to a different stable instance...

> I know this might be more of a pain to do implementation-wise

That is what I am finding at the moment. From what I can tell `:crypto.hash(:sha256, ...)` requires a binary input, so I may be missing how to avoid `term_to_binary` using the zip file? But in any case I'll monitor to see how much this negatively impacts the performance of application startup as-is.

Reply:
Though actually looking again, it may be feasible to pass down the `gtfs_timestamp` and `hastus_timestamp` as a version from `Schedule.Fetcher` and avoid the hashing entirely

Reply:
Right, sorry, I meant doing a hash in the `fetch_zip` function or something like that - but I also like your timestamp idea, that's probably even better!

Reply:
I tried out the timestamp approach in [ff15695](https://github.com/mbta/skate/pull/2198/commits/ff1569559a1b5a7f2298734641ab4ccee78afa75). I explored an alternative where instead of passing `version` to `Schedule.Data.parse_files`, the version was passed directly to `Schedule.update_state/1` as a separate prop and stored under a different key in `:persistent_term`. That would require some other reworking of reading from `:persistent_term` though since right now everything is stored under a single key, so I leaned towards preferring this version.